sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -169,38 +169,23 @@ class Column(val expr: Expression) extends Logging {
    * Returns the expression for this column either with an existing or auto assigned name.
    */
   private[sql] def named: NamedExpression = expr match {
-    // Wrap UnresolvedAttribute with UnresolvedAlias, as when we resolve UnresolvedAttribute, we
-    // will remove intermediate Alias for ExtractValue chain, and we need to alias it again to
-    // make it a NamedExpression.
-    case u: UnresolvedAttribute => UnresolvedAlias(u)
-
-    case u: UnresolvedExtractValue => UnresolvedAlias(u)
-
     case expr: NamedExpression => expr
 
     // Leave an unaliased generator with an empty list of names since the analyzer will generate
     // the correct defaults after the nested expression's type has been resolved.
     case g: Generator => MultiAlias(g, Nil)
 
-    case func: UnresolvedFunction => UnresolvedAlias(func, Some(Column.generateAlias))
-
     // If we have a top level Cast, there is a chance to give it a better alias, if there is a
     // NamedExpression under this Cast.
     case c: Cast =>
```

Comment:
can we just do `case c: Cast => UnresolvedAlias(c, Some(Column.generateAlias))`?

Reply:
Seems we can't. This code will strip the `attribute` which is below the `Cast` so I leave it.

A example, with the code, `spark.range(1).selectExpr("CAST(CAST(id as int) as double)").columns` will return the `c1` instead of `CAST(CAST(c1 as INT) as DOUBLE)`.